charts/zot/templates/tests/test-connection-fails.yaml

```diff
@@ -11,7 +11,11 @@ metadata:
 spec:
   containers:
     - name: wget
+      {{- if and .Values.test .Values.test.image }}
```

Comment:
Seems to be complaining about here?

Reply:
Third time's the charm 😄 Sorry for the back and forth.

Reply:
No worries ... we like what this PR is doing and the persistence

Still fails here ... 

Error: UPGRADE FAILED: template: zot/templates/tests/test-connection.yaml:14:41: executing "zot/templates/tests/test-connection.yaml" at <index .Values "test" "image">: error calling index: index of nil pointer

Reply:
Also, are you unable to run these tests locally? to speed this up?

Reply:
It's been passing for me locally on every commit. I think the reason it's failing in GHA is the Helm version.

Reply:
Run azure/setup-helm@v4.3.0
Installing v3.8.1
  Downloading 'v3.8.1' from 'https://get.helm.sh/'
  
  Error: UPGRADE FAILED: template: zot/templates/tests/test-connection.yaml:14:41: executing "zot/templates/tests/test-connection.yaml" at <index .Values "test" "image">: error calling index: index of nil pointer

Reply:
I cut out ~2min of waiting around, but this is working locally.

https://github.com/user-attachments/assets/3bd98982-5589-4d95-aaa4-be253c59d4d9

Reply:
To be honest, this notion of requiring backward compatibility with values.yaml for previous releases is quite unusual.

Reply:
@rchincha please run one more time so I can see the debug output from CI.

Could it be that the tests are failing in CI due to the docker rate limit? That would be ironic.

Reply:
Nvm, looks like it was indeed the Helm version. I just failed to correctly unpin it the first time around.